### PR TITLE
[BOUNTY] Fishpeople

### PR DESCRIPTION
**Pull request by ynot01**
# Not ready

# Not tested

# !!!Do not merge until IPCs are merged, then I will deal with the conflicts and such!!!

# Document the changes in your pull request

[Bounty link](https://forums.yogstation.net/threads/35-fishpeople-a-donator-race-ss13-server.22396/)

Might not satisfy the bounty terms

I'd like for this not to be restricted to a donator race

Fishpeople are a rare race of sentient fish that have taken up a specially constructed body that lets them pilot and walk around
The fish is the brain of the mob
The fish itself is organic, the body is not

The entire body has a **0.65x burn mod**, as it is water cooled & sudden bursts of temperature don't really affect the overall temperature
The entire body has a **2x cold mod**, as this is the entire body/water's temperature, it is very lethal to the fish and body
The entire body has a **2x heat mod**, as this is the entire body/water's temperature, it is very lethal to the fish and body
The entire body has a **2x shock mod**, as water is extremely conductive
The head is made of glass, and thus has a **2.5x brute mod** and will shatter upon sustaining 100 total brute damage, instantly killing the body and destroying the head, leaving only the brain (fish)
The head has full resistance to acid damage, as it is made of glass

The fishperson is paid 60% that of a human's paycheck, because it is a literal fish

Fish body is entirely mechanical, and is repaired similarly to a preternis/IPC, does not work on cryo
Fish is immune to human virus
Fish does not breath
Fish does not eat, instead it requires a steady intake of water, similar to hunger, and if thirst reaches 0% you will start taking rapid suffocation damage

# Spriting
Mob sprite TBD

Brain sprite TBD

Thirst sprite by Identification

# Wiki Documentation

# Changelog

:cl:  ynot01, Identification
rscadd: Added fishpeople
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/fishpeople.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		H.emote("gasp")
+		H.throw_alert("fish_thirst", /obj/screen/alert/fish_thirsty, 4)
+	else if(thirst < NUTRITION_LEVEL_STARVING)
+		H.throw_alert("fish_thirst", /obj/screen/alert/fish_thirsty, 3)
+	else if(thirst < NUTRITION_LEVEL_HUNGRY)
+		H.throw_alert("fish_thirst", /obj/screen/alert/fish_thirsty, 2)
+	else if(thirst < NUTRITION_LEVEL_FED)
+		H.throw_alert("fish_thirst", /obj/screen/alert/fish_thirsty, 1)
+	else
+		H.clear_alert("fish_thirst")
```
Comment (tattax) [inline code/modules/mob/living/carbon/human/species_types/fishpeople.dm]: switch switch yes yes

Comment (ynot01) [reply code/modules/mob/living/carbon/human/species_types/fishpeople.dm]: i dont think this is how switch works but this is copy pasted preternis code anyways

**Comment by Chubbygummibear** [comment]
crying in the club that fish people are going to be a fish in a suit instead of a fishperson

**Comment by ilikesaying** [comment]
lots of questions
how will the fish get a new head or get re-inserted into the body?
should the body have organs?
how does the fish get water? drinking? injection? shower?
will bodies be constructible by roboticists with research?
what stops combatants from targetting the head to near-instantly kill the body?

**Comment by Marmio64** [comment]
2.5x brute damage is pretty damn sizeable, that turns a toolbox into an energy sword and allows someone to smack them 4 times on the head with one to instantly kill them and destroy their head, a stechkin will be doing 75 damage each hit, and a single shotgun slug will deal 116 damage, which would be enough to instantly one shot them with a headshot. Should really be lowered to 2x brute damage only so its not completely devestating, but still a really sizeable downside. (And also so they don't get oneshotted by a shotgun slug to the cranium, which while realistic isn't fun). In turn for the decent buff maybe lower the burn resistance to 0.75 instead of 0.65, because honestly being able to be killed that quickly from everyday tools makes it sound like it'd be really not fun to play any time you get into a combat encounter.

**Comment by ynot01** [comment]
> lots of questions 

Lots of good questions that I appreciate and will help me keep track of my goals

> how will the fish get a new head or get re-inserted into the body? 

I plan to have fish bowls craftable either with glass in hand or easily printable, leaning towards the former as they are supposed to be easily shatterable

> should the body have organs? 

Robotic ones, yes

> how does the fish get water? drinking? injection? shower?

Mob life of the water reagent replenishes thirst, so drinking, injection, touch application, but drinking is planned to be the primarily source of replenishment; I believe shower doesn't add any reagents so that's a no

> will bodies be constructible by roboticists with research? 

No

> what stops combatants from targetting the head to near-instantly kill the body?

Nothing

**Comment by NCRLouTenant** [comment]
Ynot you absolute legend! Excited to watch this come into being.

**Comment by Firewolf34** [comment]
What happens when we put a fish into the pool? Lol!

**Comment by MenacingManatee** [comment]
How can the mob be revived after death?
Does this mob have the ability to be cloned in any way?
Can the fish (brain) be put in water to preserve it?
Will there be a manatee-themed subrace?

**Comment by ynot01** [comment]
> How can the mob be revived after death? 

Defib/cloning

> Does this mob have the ability to be cloned in any way? 

Yes

> Can the fish (brain) be put in water to preserve it? 

No

> Will there be a manatee-themed subrace?

No

**Comment by ynot01** [comment]
> What happens when we put a fish into the pool? Lol!

We'll see when IPC gets merged

**Comment by CrazyDrakkon** [comment]
> What happens when we put a fish into the pool? Lol!

Fish exits suit to take a swim. Mouse hops in bowl and steals the suit when the fish isn't looking.

**Comment by CrazyDrakkon** [comment]
> > what stops combatants from targetting the head to near-instantly kill the body?
> 
> Nothing
Researchable, somewhat sturdier plasma glass bowl that requires another set of trusted (?) hands to install?

**Comment by ynot01** [comment]
stale

**Pull request closed without merging**